src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -124,7 +124,8 @@ public BibEntry getEntry() {
 
         value = Bindings.createStringBinding(() -> {
             for (Field field : fields) {
-                if (field.getProperties().contains(FieldProperty.PERSON_NAMES)) {
+                if (field.getProperties().contains(FieldProperty.PERSON_NAMES)
+                        && nameFormatter.getValue().needsFormatting()) {
```

Comment:
You raised a good point that the formatting should be handled by the nameFormatter.
Suggestion: move the compete code of this if block to the name formatter. Then you can add the if check there, and keep the responsibility of the author formatting in that class.
For this the entry, field and database need to be passed to the `formatNameLatexFree` method (maybe pass the database already as a constructor element).